src/main.py

```
@@ -1,2 +1,17 @@
+from rdkit import Chem
+
+
 def substructure_search(mols, mol):
-    pass
\ No newline at end of file
+    matching_molecules = []
+    try:
+        substructure_mol = Chem.MolFromSmiles(mol)
+
+        for smiles in mols:
+            molecule = Chem.MolFromSmiles(smiles)
+
+            if molecule.HasSubstructMatch(substructure_mol):
+                matching_molecules.append(smiles)
+    except Exception as e:
+        print(e)
```

It is bad practice to suppress errors; in the future it will be difficult to find the source of problems